### PR TITLE
Renamings

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
This PR contains the following updates.
- Add `@Evolving` to all new classes in `api` package.
- Rename `GraphElementFrame` to `GraphElementDataset`
- Rename `NodeFrame` => `NodeDataset`
- Rename `NodeFrameBuilder` => `NodeDatasetBuilder`
- Rename `RelationshipFrame` => `RelationshipDataset`
- Rename `RelationshipFrameBuilder` => `RelationshipDatasetBuilder`
- All methods names and comments are updated accordingly.
- Change `var ds` to `val ds`
```
-final class NodeDatasetBuilder(var ds: Dataset[Row]) {
+final class NodeDatasetBuilder(val ds: Dataset[Row]) {
```

### CODE REVIEW

File: graph/api/src/main/scala/org/apache/spark/graph/api/GraphElementDataset.scala [hunk 1]
```diff
@@ -17,17 +17,17 @@
...
 
+import org.apache.spark.annotation.Evolving
 import org.apache.spark.sql.{Dataset, Row}
 
 /**
- * A [[PropertyGraph]] is created from GraphElementFrames.
- *
- * A graph element is either a node or a relationship.
- * A GraphElementFrame wraps a Dataset and describes how it maps to graph elements.
+ * A [[PropertyGraph]] whose graph element is either a node or a relationship.
```
Comment (Mats-SX) [inline graph/api/src/main/scala/org/apache/spark/graph/api/GraphElementDataset.scala:24]: It isn't exactly a `PropertyGraph`, but a building block towards that. I suggest rewording like this
```suggestion
 * A [[PropertyGraph]] component containing graph elements; either nodes or relationships.
```

File: graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala [hunk 2]
```diff
@@ -50,14 +52,14 @@ object CypherSession {
...
-   * Extracts [[NodeFrame]]s from a [[Dataset]] using column name conventions.
+   * Extracts [[NodeDataset]]s from a [[Dataset]] using column name conventions.
    *
    * For information about naming conventions, see [[CypherSession.createGraph]].
    *
    * @param nodes node dataset
    * @since 3.0.0
    */
-  def extractNodeFrames(nodes: Dataset[Row]): Set[NodeFrame] = {
+  def extractNodeDataset(nodes: Dataset[Row]): Set[NodeDataset] = {
```
Comment (Mats-SX) [inline graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala:62]: Since it produces a `Set[]`, do you think a plural form may work better?
```suggestion
  def extractNodeDatasets(nodes: Dataset[Row]): Set[NodeDataset] = {
```

File: graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala [hunk 3]
```diff
@@ -88,19 +90,19 @@ object CypherSession {
...
-   * Extracts [[RelationshipFrame]]s from a [[Dataset]] using column name conventions.
+   * Extracts [[RelationshipDataset]]s from a [[Dataset]] using column name conventions.
    *
    * For information about naming conventions, see [[CypherSession.createGraph]].
    *
    * @param relationships relationship dataset
    * @since 3.0.0
    */
-  def extractRelationshipFrames(relationships: Dataset[Row]): Set[RelationshipFrame] = {
+  def extractRelationshipDataset(relationships: Dataset[Row]): Set[RelationshipDataset] = {
```
Comment (Mats-SX) [inline graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala:105]: Same as above regarding plural form
```suggestion
  def extractRelationshipDatasets(relationships: Dataset[Row]): Set[RelationshipDataset] = {
```

File: graph/api/src/main/scala/org/apache/spark/graph/api/NodeDatasetBuilder.scala [hunk 4]
```diff
@@ -19,15 +19,17 @@ package org.apache.spark.graph.api
...
 /**
- * Interface used to build a [[NodeFrame]].
+ * A builder for [[NodeDataset]].
  *
  * @param ds Dataset containing a single node in each row
  * @since 3.0.0
  */
-final class NodeFrameBuilder(var ds: Dataset[Row]) {
+@Evolving
+final class NodeDatasetBuilder(val ds: Dataset[Row]) {
```
Comment (Mats-SX) [inline graph/api/src/main/scala/org/apache/spark/graph/api/NodeDatasetBuilder.scala:32]: Good catch on the `val`

**Review by Mats-SX — APPROVED** [review]
I have a few comments that we can discuss, but I'll merge this into the Spark PR branch pre-emptively. We can apply my comments later if you agree.

**Comment by dongjoon-hyun** [comment]
Thank you for merging, @Mats-SX . Yes. Please update them in the main PR~ I agree.
